static/js/message_scroll.js

```diff
@@ -60,17 +60,31 @@ exports.scroll_finished = function () {
 };
 
 var scroll_timer;
-function scroll_finish() {
+function scroll_finish(opts) {
     actively_scrolling = true;
     clearTimeout(scroll_timer);
-    scroll_timer = setTimeout(exports.scroll_finished, 100);
+
+    function finish() {
+        exports.scroll_finished(opts);
+    }
+
+    // TODO: consider removing the 100ms timeout here, since
+    //       we are already on a 50ms throttle
+    scroll_timer = setTimeout(finish, 100);
```

Comment:
I think the 100ms timeout here is potentially correct; we want to post 100ms after you finish a sequence of scrolling actions, and we only want to check every 50ms because checking consumes some resources.